cluster/storage/src/main/java/com/linkedin/openhouse/cluster/storage/Storage.java

```diff
@@ -0,0 +1,45 @@
+package com.linkedin.openhouse.cluster.storage;
+
+import java.util.Map;
+
+/**
+ * The Storage interface represents a storage system in OpenHouse. It provides methods to check if
+ * the storage is active, retrieve properties of the storage, get the type of the storage, and get a
+ * client to interact with the storage.
+ *
+ * <p>Implementations of this interface should provide the specific logic for each type of storage.
+ * For example, the {@link com.linkedin.openhouse.cluster.storage.local.LocalStorage} class is an
+ * implementation of this interface for local storage, and it uses a {@link
+ * com.linkedin.openhouse.cluster.storage.local.LocalStorageClient} to interact with the local file
+ * system.
+ */
+public interface Storage {
+
+  /**
+   * Check if the storage is active.
+   *
+   * @return true if the storage is active, false otherwise
+   */
+  boolean isActive();
```

Comment:
Storage is considered active if cluster-yaml has property defined for it.
for example in the end-state if we have cluster-yaml **only** as:
```
storages:
   hdfs: 
           endpoint:
           ..
```
then `HDFSStorage.isActive()=true` and `S3Storage.isActive=false`

it will be used by StorageManager to supply a Storage. Lemme know if its not clear, can raise a subsequent PR.

Reply:
see the usage here: https://github.com/HotSushi/openhouse/pull/1/files#diff-3eab08eaf8a381dc4108e6db8d27ff2f34a126442a904758bb0f5ca9211c54bdR74

Reply:
👍

Reply:
I had the same question. May be adda. comment describing what active means?
Do you think active means that storage is configured? May be isConfigured() is a better name?

Reply:
yes! isConfigured is definitily a better name, will address

Reply:
Addressed